### PR TITLE
fix: paths /, //, generate an ArrayIndexOutOfBoundException

### PR DESCRIPTION
**Pull request by lgw-gravitee**
applicable with vhost path="/"
fixes [AM-6936]

[AM-6936]: https://gravitee.atlassian.net/browse/AM-6936?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/vertx/core/http/VertxHttpServerRequest.java [hunk 1]
```diff
@@ -77,6 +78,16 @@ public VertxHttpServerRequest(HttpServerRequest httpServerRequest) {
...
 
+    private static String extractContextPath(HttpServerRequest httpServerRequest) {
+        if(httpServerRequest.path() == null) {
+            return null;
+        } else {
+            return Stream.of(httpServerRequest.path().split("/"))
+                    .findFirst()
+                    .orElse(null);
+        }
+    }
```
[the remark this answers is not in the document]
Comment (lgw-gravitee) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/vertx/core/http/VertxHttpServerRequest.java:89]: when path.isEmpty it should behave as previously 
`httpServerRequest.path() != null ? httpServerRequest.path().split("/")[0] : null` -> `""`

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/test/java/io/gravitee/am/gateway/handler/common/vertx/core/http/VertxHttpServerRequestTest.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    @Test
+    public void shouldReturnNullContextPathWhenPathIsRoot() {
+        assertNull(new VertxHttpServerRequest(mockHttpServerRequest("/")).contextPath());
+        assertNull(new VertxHttpServerRequest(mockHttpServerRequest("//")).contextPath());
+        assertNull(new VertxHttpServerRequest(mockHttpServerRequest("///")).contextPath());
+    }
+
+    @Test
+    public void shouldReturnContextPathWhenPathIsEmpty() {
```
Comment (wajb) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/test/java/io/gravitee/am/gateway/handler/common/vertx/core/http/VertxHttpServerRequestTest.java:39]: suggestion: add a nominal test case for a non-empty context path.

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/vertx/core/http/VertxHttpServerRequest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (wajb) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/vertx/core/http/VertxHttpServerRequest.java:89]: Would a value of "" not trigger the same reported out of bounds exception? 🤔

[the remark this answers is not in the document]
Comment (wajb) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/vertx/core/http/VertxHttpServerRequest.java:89]: Never mind, I understand that this is a valid fix.

**Review by wajb — APPROVED** [review]
